### PR TITLE
Fix Identify tool on virtual point cloud layer

### PR DESCRIPTION
**Pull request by dvdkon**
## Description

This PR fixes the Identify map tool on virtual point cloud (VPC) layers, since it currently doesn't display any points from these layers.

This PR changes two methods:

- `QgsPointCloudRenderer::identify()` not to rely on the main index, since it is missing for VPC layers.
-  `QgsPointCloudDataProvider::identify()` to consider points in sub-indexes.

Will fix #54729

### CODE REVIEW

File: src/core/pointcloud/qgspointclouddataprovider.cpp [hunk 1]
```diff
@@ -311,13 +311,36 @@ QVector<QVariantMap> QgsPointCloudDataProvider::identify(
...
+  QVector<QVariantMap> acceptedPoints;
+
+  // Try sub-indexes first
+  for ( QgsPointCloudSubIndex &subidx : subIndexes() )
+  {
+    // Check if the sub-index is relevant and if it is loaded. We shouldn't
+    // need to identify points in unloaded indices.
+    if ( !subidx.index()
+         || !subidx.polygonBounds().intersects( extentGeometry )
+         || ( !subidx.zRange().overlaps( extentZRange ) ) )
```
Comment (nyalldawson) [inline src/core/pointcloud/qgspointclouddataprovider.cpp:324]: Let's do the (considerably!) cheaper check first:

```suggestion
         || ( !subidx.zRange().overlaps( extentZRange ) ) )
         || !subidx.polygonBounds().intersects( extentGeometry )
```

Comment (dvdkon) [reply src/core/pointcloud/qgspointclouddataprovider.cpp]: Done.

**Comment by nyalldawson** [comment]
@dvdkon the new test is failing here, can you take a look?

**Review by wonder-sk — APPROVED** [review]

**Comment by wonder-sk** [comment]
Only unrelated `test_gui_processinggui` test failing...

**Pull request merged**